airflow/providers/sqlite/hooks/sqlite.py

```diff
@@ -33,13 +34,20 @@ class SqliteHook(DbApiHook):
 
     def get_conn(self) -> sqlite3.dbapi2.Connection:
         """Returns a sqlite connection object"""
-        conn_id = getattr(self, self.conn_name_attr)
-        airflow_conn = self.get_connection(conn_id)
-        conn = sqlite3.connect(airflow_conn.host)
+        # The sqlite3 connection does not use the sqlite prefix.
+        # See https://docs.sqlalchemy.org/en/14/dialects/sqlite.html#connect-strings for details.
+        uri = self.get_uri().replace("sqlite:///", "file:")
+        conn = sqlite3.connect(uri, uri=True)
```

Comment:
Is `:memory:` a viable value here? This logic does not support that.

Reply:
Yes, it does. For example, see the last example [here](https://docs.python.org/3/library/sqlite3.html#how-to-work-with-sqlite-uris).
> db = "file:mem1?mode=memory&cache=shared"
> con1 = sqlite3.connect(db, uri=True)

Reply:
More examples can be found [here](https://www.sqlite.org/inmemorydb.html).

Reply:
So you can’t use the `sqlite://` string (no way to configure a connection that formats into this specific string), but same effect can be achieved using the URL format, right? I wonder if we want to document this somewhere.

Reply:
You can. The host field is optional. You can create an empty SQLite connection and it will use an in-memory db.

Reply:
I have added a test case for that 37d2a53.

And validated that it works locally:
```
>>> import sqlite3
>>> sqlite3.connect("file:")
<sqlite3.Connection object at 0xffffb71a4d50>
```